config/errors.go

```diff
@@ -91,3 +103,50 @@ type ErrInvalidURIPrefix string
 func (e ErrInvalidURIPrefix) Error() string {
 	return fmt.Sprintf("config: invalid uri_prefix (%v). uri_prefix must start with a forward slash '/' ", string(e))
 }
+
+// ErrUnknownProviderType is return when the config contains a provider type that has not been registered
+type ErrUnknownProviderType struct {
+	Name           string // name of the entry in the config
+	Type           string // name of the data provider
```

Comment:
name -> type

Reply:
The comment is indicating that `Type` is really the name of the data provider
And that Name here is the anme of the entry in the config. 
I'll change the comments to the following:
// Name is the name of the entry in the config
// Type is the name of the data provider